File: Lista_Logica_Programacao/Python/ex8.py
# Exercício 8 - Jogo de Adivinhação

import random  # usamos a biblioteca random para sortear números

# sorteio do número entre 1 e 100 (inclusive)
numero = random.randint(1, 100)

# contador de tentativas começa em zero
tentativas = 0

# loop while para repetir até o usuário acertar
while True:
    # pede palpite e converte para int porque input é texto
    palpite = int(input("Digite um palpite (1-100): "))
    tentativas += 1  # cada interação conta uma tentativa

    # verifica se acertou ou dá dica
    if palpite == numero:
        print(f"Parabéns! Você acertou em {tentativas} tentativas.")
        break  # sai do loop quando acerta
    elif palpite < numero:
        print("Tente um número maior.")  # dica: chute maior
    else:
        print("Tente um número menor.")  # dica: chute menor

# Anotações:
# - usei random.randint(1, 100) porque ele devolve um inteiro aleatório entre os limites.
# - while True cria um laço infinito que só termina com break quando o palpite está correto.
# - contador de tentativas é importante para saber quantas jogadas o usuário fez.
# - if/elif/else servem para comparar o palpite com o número e dar dica de "maior" ou "menor".
# - converti o input para int porque input sempre retorna string, e precisamos comparar números.
# - coloquei comentários explicando cada parte, igual às outras atividades, para revisar depois e explicar ao professor.

File: Lista_Logica_Programacao/Java/Ex8.java
// Exercício 8 - Jogo de Adivinhação

import java.util.Random;
import java.util.Scanner;

public class Ex8 {
    public static void main(String[] args) {
        Random rand = new Random();
        Scanner sc = new Scanner(System.in);

        // sorteia número entre 1 e 100
        int numero = rand.nextInt(100) + 1;
        int tentativas = 0;

        while (true) {
            System.out.print("Digite um palpite (1-100): ");
            int palpite = sc.nextInt();
            tentativas++;

            if (palpite == numero) {
                System.out.println("Parabéns! Você acertou em " + tentativas + " tentativas.");
                break;
            } else if (palpite < numero) {
                System.out.println("Tente um número maior.");
            } else {
                System.out.println("Tente um número menor.");
            }
        }

        sc.close();
    }
}
/*
Anotações:

- Usei Random e Scanner; não esqueço de importar.
- rand.nextInt(100) retorna 0..99, por isso somo 1 para ficar 1..100.
- while(true) repete até o break quando acerta.
- contador de tentativas incrementa a cada laço.
- if/else com dicas maior/menor igual ao programa em Python.
- Scanner é fechado no final, sempre.
*/